diff -r -U 3 minecraft_server\net\minecraft\item\ItemMap.java minecraft_server_patched\net\minecraft\item\ItemMap.java
--- minecraft_server\net\minecraft\item\ItemMap.java
+++ minecraft_server_patched\net\minecraft\item\ItemMap.java
@@ -137,7 +137,7 @@
                            i5 = 0;
                         }
 
-                        MapColor mapcolor = (MapColor)Iterables.getFirst(Multisets.<T>copyHighestCountFirst(multiset), MapColor.field_151660_b);
+                        MapColor mapcolor = (MapColor)Iterables.getFirst(Multisets.<MapColor>copyHighestCountFirst(multiset), MapColor.field_151660_b);
                         if(mapcolor == MapColor.field_151662_n) {
                            d2 = (double)k3 * 0.1D + (double)(k1 + l1 & 1) * 0.2D;
                            i5 = 1;
