diff -r -U 3 minecraft_server\net\minecraft\util\MessageSerializer.java minecraft_server_patched\net\minecraft\util\MessageSerializer.java
--- minecraft_server\net\minecraft\util\MessageSerializer.java
+++ minecraft_server_patched\net\minecraft\util\MessageSerializer.java
@@ -37,9 +37,9 @@
          packetbuffer.func_150787_b(integer.intValue());
 
          try {
-            if(p_encode_2_ instanceof S0CPacketSpawnPlayer) {
-               p_encode_2_ = p_encode_2_;
-            }
+//            if(p_encode_2_ instanceof S0CPacketSpawnPlayer) {
+//               p_encode_2_ = p_encode_2_;
+//            }
 
             p_encode_2_.func_148840_b(packetbuffer);
          } catch (Throwable throwable) {
