diff -r -U 3 minecraft_server\net\minecraft\world\gen\structure\StructureVillagePieces.java minecraft_server_patched\net\minecraft\world\gen\structure\StructureVillagePieces.java
--- minecraft_server\net\minecraft\world\gen\structure\StructureVillagePieces.java
+++ minecraft_server_patched\net\minecraft\world\gen\structure\StructureVillagePieces.java
@@ -29,6 +29,7 @@
 import net.minecraft.world.gen.structure.StructureBoundingBox;
 import net.minecraft.world.gen.structure.StructureComponent;
 
+@SuppressWarnings("incomplete-switch")
 public class StructureVillagePieces {
    public static void func_143016_a() {
       MapGenStructureIO.func_143031_a(StructureVillagePieces.House1.class, "ViBH");
