diff -r -U 3 minecraft\net\minecraft\block\BlockTripWireHook.java minecraft_patched\net\minecraft\block\BlockTripWireHook.java
--- minecraft\net\minecraft\block\BlockTripWireHook.java
+++ minecraft_patched\net\minecraft\block\BlockTripWireHook.java
@@ -197,6 +197,7 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    public void func_180654_a(IBlockAccess p_180654_1_, BlockPos p_180654_2_) {
       float f = 0.1875F;
       switch((EnumFacing)p_180654_1_.func_180495_p(p_180654_2_).func_177229_b(field_176264_a)) {
