diff -r -U 3 minecraft\net\minecraft\client\network\NetHandlerPlayClient.java minecraft_patched\net\minecraft\client\network\NetHandlerPlayClient.java
--- minecraft\net\minecraft\client\network\NetHandlerPlayClient.java
+++ minecraft_patched\net\minecraft\client\network\NetHandlerPlayClient.java
@@ -1112,6 +1112,7 @@
       p_175093_1_.func_179788_a(this.field_147300_g.func_175723_af());
    }
 
+   @SuppressWarnings("incomplete-switch")
    public void func_175099_a(S45PacketTitle p_175099_1_) {
       PacketThreadUtil.func_180031_a(p_175099_1_, this, this.field_147299_f);
       S45PacketTitle.Type s45packettitle$type = p_175099_1_.func_179807_a();
@@ -1155,6 +1156,7 @@
 
    }
 
+   @SuppressWarnings("incomplete-switch")
    public void func_147256_a(S38PacketPlayerListItem p_147256_1_) {
       PacketThreadUtil.func_180031_a(p_147256_1_, this, this.field_147299_f);
 
