diff -r -U 3 minecraft\net\minecraft\client\gui\GuiOverlayDebug.java minecraft_patched\net\minecraft\client\gui\GuiOverlayDebug.java
--- minecraft\net\minecraft\client\gui\GuiOverlayDebug.java
+++ minecraft_patched\net\minecraft\client\gui\GuiOverlayDebug.java
@@ -92,6 +92,7 @@
 
    }
 
+   @SuppressWarnings("incomplete-switch")
    protected List<String> call() {
       BlockPos blockpos = new BlockPos(this.field_175242_a.func_175606_aa().field_70165_t, this.field_175242_a.func_175606_aa().func_174813_aQ().field_72338_b, this.field_175242_a.func_175606_aa().field_70161_v);
       if(this.func_175236_d()) {
